### DISCUSSION TITLE
How to move a cookiecutter based package to a copier package ?

### DISCUSSION THREAD

**Discussion by 12rambau** [copier-org/copier#1393] [Q&A] [answered]
I tried to find this question in several forums but I guess my wording was wrong as I didn't find anything. I'm positive I'm not the first one facing this issue and maybe there are no solutions. 

I was maintaining a `cookiecutter`  based template for my Python packages (https://github.com/12rambau/pypackage). As I used it extensively for my developments and as I'm now facing the classic issue of "my packages are getting out of sync with Python best practices what should I do ?" I decided to move it to `copier`.

Now what I would love to do is wire the existing projects to the copier template so that they become easy to update. Is it even possible? 
Note that I'm willing to go over complex manual processes if needed as it's a 1 time change and it will save me countless hours of stupid work.

**Comment by pawamoy** [comment, marked as answer]
What I'd do:

- generate the existing project again, from the Copier template
- copy/move existing contents to this freshly generated project (the most difficult part), including your `.git` folder
- add and commit everything
- your project is now tracked by Copier :slightly_smiling_face:

**Reply by 12rambau** [reply]
The trick is to move the .git folder to keep track of everything, perfect, Let's see how it goes !

**Comment by slocoro** [comment]
Is it not possible to manually create `.copier-answer.yml` file with the same answers provided in the original cookecutter template and adding the `_commit` key with the commit/tag of the new copier template?

**Reply by pawamoy** [reply]
Well feel free to try. If both templates are very similar, it could work I think.

**Reply by 12rambau** [reply]
I did it several time and it works, it's just hacky I highly suggest to duplicate your local git repository to avoid breaking when you'll run copier update.

**Reply by slocoro** [reply]
@12rambau So you think the other approach (i.e. generating project from new template and copying contents) works better manually creating a `.copier-answer.yml` file?

**Comment by yajo** [comment]
For Copier, a project built from a cookiecutter template is no different from any other project. Copier even understands that a project can have various templates applied. So, applying a template to a preexisting project is perfectly supported. Just:

1. `copier copy src dest`
3. Answer questions
4. Do manual adjustments. Probably it will involve discarding most diff. A merge tool such as meld or vSCode will help.
5. Commit.
